### DISCUSSION TITLE
Customizable redirect page

### DISCUSSION THREAD

**Discussion by GioPan04** [withastro/roadmap#863] [Proposal]
# Summary

Add the ability to customize the redirect page generated in SSG mode

# Background & Motivation

Currently when the user is redirected in a website built with SSG mode on it's presented with a blank page with a non customizable redirect timeout and text.

For example: using i18n if a page is missing in a secondary language, the user is redirected to the same page in the fallback language, but I don't have a way to tell the user that he has been redirected to that page because the original page wasn't available in his language.

It would be useful if we have the opportunity to create a custom page that is used as a template for redirects. So we could even change the redirect timeout easily 

# Goals

- Have a way to customize generated redirect pages for SSG
- Per locales template would be better, so that we could translate even the redirect page

# Example

The redirect template would be something like this:
```astro
---
interface Props {
  fromUrl: string;
  toUrl: string;
}

const { fromUrl, toUrl } = Astro.props;
---
<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="refresh" content={`10; url=${toUrl}`} />
    <meta name="viewport" content="width=device-width" />
    <meta name="generator" content={Astro.generator} />
  </head>
  <body class="...">
    <main class="...">
      <span>You have been redirected because the page you were visiting wasn't available in your language</span>
      <span>You'll be redirected to the English version of the website</span>
    </main>
  </body>
</html>
```

And we could pass the template path in the config file: 
```js
export default defineConfig({
  integrations: [...],
  site: "...",
  i18n: {
    defaultLocale: "en",
    locales: ["en", "it"],
    fallback: {
      "it": {
        language: "en",
        redirectLayout: "..."
      }
    }
  }
});
```

**Comment by florian-lefebvre** [comment]
Duplicate of #844
